.github/workflows/test-and-publish-on-push-to-main.yml

```diff
@@ -0,0 +1,96 @@
+name: Test (and Publish to Docker Hub on push to main)
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+    - main
+    - github-actions
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        flavor:
+          - octodns
+          - azure
+          - cloudflare
+          - constellix
+          - digitalocean
+          - dnsimple
+          - dnsmadeeasy
+          - dyn
+          - easydns
+          - edgedns
+          - etchosts
+          - gandi
+          - gcore
+          - googlecloud
+          - hetzner
+          - mythicbeasts
+          - ns1
+          - ovh
+          - powerdns
+          - rackspace
+          - route53
+          - selectel
+          - transip
+          - ultra
+    steps:
+      - uses: actions/checkout@v3
+      - name: set up qemu virt with platforms
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: arm64,arm
+      - name: set up docker buildx builder with platforms
+        uses: docker/setup-buildx-action@v2
+        with:
+          platforms: linux/amd64,linux/arm64,linux/arm/v7
+      - name: login to the Docker registry
```

Comment:
Let's wait to login until we're about to push so that this can run for contributors running on their own forks.